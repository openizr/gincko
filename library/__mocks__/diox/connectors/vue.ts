/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

/**
 * diox/connectors/vue mock.
 */
const connect = vi.fn(() => vi.fn(() => ({
  steps: [{ id: 'start' }, { id: 'end' }],
  loading: process.env.LOADING === 'true',
})));

export default connect;
