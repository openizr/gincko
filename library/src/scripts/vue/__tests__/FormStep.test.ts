/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @vitest-environment jsdom
 */

import { render } from '@testing-library/vue';
import FormStep from 'scripts/vue/FormStep.vue';

vi.mock('biuty/vue');
vi.mock('scripts/core/Engine');
vi.mock('scripts/vue/FormField.vue');

describe('vue/FormStep', () => {
  const customComponents = {};
  const userInputs = { test: 'ok' };
  const variables = { var: 'value' };
  const i18n = vi.fn(() => 'TRANSLATED LABEL');
  const onUserAction = vi.fn() as unknown as OnUserAction;

  beforeEach(() => {
    vi.clearAllMocks();
    Object.assign(console, { warn: vi.fn() });
  });

  test('renders correctly - active step', async () => {
    const { container } = render(FormStep, {
      props: {
        index: 0,
        isActive: true,
        step: {
          id: 'step',
          status: 'success',
          fields: [{
            id: 'Message',
            status: 'initial',
            component: 'Message',
            componentProps: {},
          }],
        },
        i18n,
        variables,
        userInputs,
        onUserAction,
        customComponents,
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - inactive step', async () => {
    const { container } = render(FormStep, {
      props: {
        index: 0,
        step: {
          id: 'step',
          status: 'success',
          fields: [{
            id: 'Message',
            status: 'initial',
            component: 'Message',
            componentProps: {},
          }, null],
        },
        i18n,
        variables,
        userInputs,
        onUserAction,
        isActive: false,
        customComponents,
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
