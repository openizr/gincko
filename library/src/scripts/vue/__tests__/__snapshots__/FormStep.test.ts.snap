// Vitest Snapshot v1

exports[`vue/FormStep > renders correctly - active step 1`] = `
<div
  class="gincko__step success step active"
  id="step__0"
>
  <div
    class="gincko__step__fields"
  >
    <!-- Key is composed of both step and field ids, in order to ensure each field is correctly
      reset when user changes his journey in previous steps. -->
    
    <div
      id="field"
    >
      {"field":{"id":"Message","status":"initial","component":"Message","componentProps":{}},"path":"step.0.Message","isActive":true,"variables":{"var":"value"},"userInputs":{"test":"ok"},"customComponents":{}}
    </div>
    
  </div>
</div>
`;

exports[`vue/FormStep > renders correctly - inactive step 1`] = `
<div
  class="gincko__step success step"
  id="step__0"
>
  <div
    class="gincko__step__fields"
  >
    <!-- Key is composed of both step and field ids, in order to ensure each field is correctly
      reset when user changes his journey in previous steps. -->
    
    <div
      id="field"
    >
      {"field":{"id":"Message","status":"initial","component":"Message","componentProps":{}},"path":"step.0.Message","isActive":false,"variables":{"var":"value"},"userInputs":{"test":"ok"},"customComponents":{}}
    </div>
    
  </div>
</div>
`;
